ArcGISRuntimeSDKQt_CppSamples/Maps/GenerateOfflineMap_Overrides/GenerateOfflineMap_Overrides.cpp

```diff
@@ -0,0 +1,458 @@
+
+GenerateOfflineMap_Overrides::GenerateOfflineMap_Overrides(QQuickItem* parent /* = nullptr */):
+  QQuickItem(parent)
+{
+}
+
+void GenerateOfflineMap_Overrides::init()
+{
+  // Register the map view for QML
+  qmlRegisterType<MapQuickView>("Esri.Samples", 1, 0, "MapView");
+  qmlRegisterType<GenerateOfflineMap_Overrides>("Esri.Samples", 1, 0, "GenerateOfflineMap_OverridesSample");
+  qmlRegisterUncreatableType<AuthenticationManager>("Esri.Samples", 1, 0, "AuthenticationManager", "AuthenticationManager is uncreateable");
+}
+
+void GenerateOfflineMap_Overrides::componentComplete()
+{
+  QQuickItem::componentComplete();
+
+  // find QML MapView component
+  m_mapView = findChild<MapQuickView*>("mapView");
+
+  // Create a Portal Item for use by the Map and OfflineMapTask
+  bool loginRequired = true;
+  Portal* portal = new Portal(loginRequired, this);
+  m_portalItem = new PortalItem(portal, webMapId(), this);
+
+  // Create a map from the Portal Item
+  m_map = new Map(m_portalItem, this);
+
+  // Update property once map is done loading
+  connect(m_map, &Map::doneLoading, this, [this](Error e)
+  {
+    if (!e.isEmpty())
+      return;
+
+    m_mapLoaded = true;
+    emit mapLoadedChanged();
+  });
+
+  // Set map to map view
+  m_mapView->setMap(m_map);
+
+  // Create the OfflineMapTask with the PortalItem
+  m_offlineMapTask = new OfflineMapTask(m_portalItem, this);
+
+  // connect to the error signal
+  connect(m_offlineMapTask, &OfflineMapTask::errorOccurred, this, [](Error e)
+  {
+    if (e.isEmpty())
+      return;
+
+    qDebug() << e.message() << e.additionalMessage();
+  });
+
+  // connect to the signal for when the default parameters are generated
+  connect(m_offlineMapTask, &OfflineMapTask::createDefaultGenerateOfflineMapParametersCompleted,
+          this, [this](QUuid, const GenerateOfflineMapParameters& params)
+  {
+    // Use the parameters to create a set of overrides.
+    m_parameters = params;
+    m_offlineMapTask->createGenerateOfflineMapParameterOverrides(params);
+  });
+
+  connect(m_offlineMapTask, &OfflineMapTask::createGenerateOfflineMapParameterOverridesCompleted,
+          this, [this](QUuid /*id*/, GenerateOfflineMapParameterOverrides* parameterOverrides)
+  {
+    m_parameterOverrides = parameterOverrides;
+    emit overridesReadyChanged();
+    m_taskBusy = false;
+    emit taskBusyChanged();
+  });
+}
+
+void GenerateOfflineMap_Overrides::setAreaOfInterest(double xCorner1, double yCorner1, double xCorner2, double yCorner2)
+{
+  // create an envelope from the QML rectangle corners
+  const Point corner1 = m_mapView->screenToLocation(xCorner1, yCorner1);
+  const Point corner2 = m_mapView->screenToLocation(xCorner2, yCorner2);
+  const Envelope extent(corner1, corner2);
+  const Envelope mapExtent = GeometryEngine::project(extent, SpatialReference::webMercator());
+
+  // generate parameters
+  m_offlineMapTask->createDefaultGenerateOfflineMapParameters(mapExtent);
+  m_taskBusy = true;
```

Comment:
might be worth factoring out a private `setBusy(bool busy)` method that takes care of updating the value and emitting the signal